blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -147,10 +147,6 @@ sealed abstract class BlazeClientBuilder[F[_]] private (
   def withDefaultSslContext: BlazeClientBuilder[F] =
     withSslContext(SSLContext.getDefault())
 
-  /** Use some provided `SSLContext` when making secure calls, or disable secure calls with `None` */
-  def withSslContextOption(sslContext: Option[SSLContext]): BlazeClientBuilder[F] =
```

Comment:
Per my commit's message, if `SSLContextOption` is truly going to be internal, then I think it's worthwhile to reduce its accessibility via `private [X] ...`. Thoughts, @rossabaker?

Reply:
I guess I'd leave this one deprecated and fold it into `NoSSL` and `Provided`, but I've been critcized for overusing deprecation. :laughing: 

But I think it's legimtiate.  This is an old implementation detail that leaked, but it doesn't cost us anything to preserve the old with a note, so anyone calling it has a gentle path to the new.

Reply:
> I guess I'd leave this one deprecated and fold it into NoSSL and Provided

👍

> This is an old implementation detail that leaked, but it doesn't cost us anything to preserve the old with a note,

 👍

Reply:
FWIW I am very much in favor of deprecation if it leads to a discoverable migration path. 👍

Reply:
I'm glad I'm not the only one who likes that approach. :smile:

Reply:
At my last job I was a build engineer. The thing that ended up being a big self-caused problem was that the initial way I set up the builds, while a vast overall improvement on what they'd had before, was not as discoverable as it should have been. As a result, on my second iteration I had a rule for myself that when I went to manually test things, I would start with a completely empty scenario---something where I had none of the build dependencies/files in place that were necessary for the build to succeed, and then I would walk through the whole process of setting it up by making sure that the errors that were given me were sufficient for me to fix the problem from the information provided in them alone. It actually turned out to be a very useful exercise in reflecting on what I had built and it even helped me as a user when I forgot to do something (because yes sometimes I was the "dumb" user). Since then, I've had a strong belief in discoverability. I spent too many hours explaining and re-explaining the same things to developers because of my first iteration. I decided that my time was better spent up front in improving the UX rather than down the road in doing support.

Reply:
> I decided that my time was better spent up front in improving the UX rather than down the road in doing support.

💯